### PR TITLE
GITHUB-5307-1_7

### PR DESCRIPTION
**Pull request by a2xchip**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

#5307 

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | OK
| Added integration tests           | -
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: features/Context/WebUser.php [hunk 1]
```diff
@@ -842,6 +842,42 @@ public function iChangeTheTo($field, $value = null, $language = null)
...
+    public function iShouldSeeSelectChoicesOrdered($field, PyStringNode $items)
+    {
+        $searched = array_values(explode(',', implode(',', $items->getLines())));
+        $foundChoices = $this->getCurrentPage()
+            ->find(
+                'css',
+                sprintf('label:contains("%s")', $field)
+            )
+            ->getParent()
+            ->getParent()
```
Comment (pierallard) [inline features/Context/WebUser.php]: Prefer use ClosestTrait to avoid getParent() n times

File: features/Context/WebUser.php [hunk 2]
```diff
@@ -842,6 +842,42 @@ public function iChangeTheTo($field, $value = null, $language = null)
...
+
+        $fieldsArray = [];
+        foreach ($foundChoices as $choice) {
+            $fieldsArray[] = trim($choice->getText());
+        }
+
+        $fieldsArray = array_values(array_filter($fieldsArray));
+
+        if ($searched !== $fieldsArray) {
+            var_dump($fieldsArray, $searched);
```
Comment (pierallard) [inline features/Context/WebUser.php]: Nop!

File: features/Context/WebUser.php (hunk 1 above)
Comment (a2xchip) [inline features/Context/WebUser.php]: @pierallard unfortunatelly closest trait does not helps here. Got:
```
 invalid selector: The result of the xpath expression "(((((((((((((((((//html/descendant-or-self::label[contains(string(.), 'Attribute group')])[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/..)[1]/.." is: [object HTMLDocument]. It should be an element.
        (Session info: chrome=56.0.2924.87)
```

Comment (momoss) [inline features/Context/WebUser.php]: Maybe something like that could work:
return $this->spin(function () use ($field) {
$labels = $this->getCurrentPage()->find('css', sprintf('label:contains("%s")', $field));
if ($field !== null) {
      return $this->getClosest($field, 'field-input');
}

return null;
}, sprintf('Can not find .... "%s"', $field));

Comment (a2xchip) [inline features/Context/WebUser.php]: Gonna try later! Thx

**Comment by momoss** [comment]
Please don't forget the CHANGELOG.md

**Comment by pierallard** [comment]
Update CHANGELOG then PR can be merged.

**Comment by a2xchip** [comment]
@pierallard No NOoooo!!! Do not merge problem is not in CHANGELOG! We decided to extract ordering in repo method!!!

**Comment by pierallard** [comment]
@a2xchip We do exactly this type of thing in at least 10 forms. Not a problem.

**Comment by a2xchip** [comment]
Wow! May be could be optimized somehow @pierallard

**Comment by pierallard** [comment]
CI run on https://core-ci.akeneo.com/blue/organizations/jenkins/akeneo%2Fpim-community-dev/detail/PR-5814/17/pipeline/

**Pull request merged**